packages/babel-plugin-transform-typescript/src/index.js

```diff
@@ -42,17 +42,25 @@ export default function() {
         let allElided = true;
         const importsToRemove: Path<Node>[] = [];
 
-        for (const specifier of path.node.specifiers) {
+        for (let i = 0; i < path.node.specifiers.length; i++) {
+          const specifier = path.node.specifiers[i];
           const binding = path.scope.getBinding(specifier.local.name);
 
           // The binding may not exist if the import node was explicitly
           // injected by another plugin. Currently core does not do a good job
           // of keeping scope bindings synchronized with the AST. For now we
           // just bail if there is no binding, since chances are good that if
           // the import statement was injected then it wasn't a typescript type
-          // import anyway.
+          // import anyway
           if (binding && isImportTypeOnly(binding, state.programPath)) {
-            importsToRemove.push(binding.path);
+            // Binding points to originally existing import specifier. If
+            // another plugin replaces original import declaration with custom
+            // one, removing bindings won't change this declaration
+            const specifierPath = path.get(`specifiers.${i}`);
```

Comment:
I don't understand how this works? How does a string like "specifiers.2" get inserted into the path?

Reply:
Hmm I guess it might be an array so it's actually `specifiers[i]` but the path version?

Reply:
As I understood [babel-handbook](https://github.com/thejameskyle/babel-handbook/blob/master/translations/en/plugin-handbook.md#get-the-path-of-sub-node), it is a method to get a path of sub-node. If I use `specifiers[i]`, it will return instance of `Node`, but I need `Path`.

Reply:
Will accessing this by index still work if a plugin shuffled the imports around?
For that matter, what if a plugin changed the code so that an imported variable which used to be used only as a type is now used as a value? It doesn't seem like we can be resilient in the case of plugins doing arbitrary things, unless we get a fresh binding.

Reply:
@andy-ms I'm actually testing it with plugin that changes import ([elide-with-previous-change](https://github.com/Lodin/babel/blob/f7aca4b93bc8380a189b115f87b5110c00d3cfdf/packages/babel-plugin-transform-typescript/test/fixtures/imports/elide-with-previous-change/plugin.js)), because I found the problem creating this kind of plugin.

Unfortunately, cannot say anything about the situation you mentioned. Most likely the value will be removed. 

I think if there is a way to refresh bindings before start, it would be the best approach. Updating bindings will update paths in them which makes my changes redundant.